dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/metrics/ProcessInstanceMetrics.java

```diff
@@ -31,6 +36,24 @@ private ProcessInstanceMetrics() {
         throw new UnsupportedOperationException("Utility class");
     }
 
+    private static Map<String, Counter> PROCESS_INSTANCE_COUNTERS = new HashMap<>();
+
+    private static final Set<String> PROCESS_INSTANCE_STATES = ImmutableSet.of(
+            "submit", "timeout", "finish", "failover", "success", "fail", "stop");
```

Comment:
It's better to directly use `ExecutionStatus.SUCCESS.getCode()` to replace the `success`, and use 
```java
ProcessInstanceMetrics.incProcessInstanceByState(ExecutionStatus.SUCCESS.getCode());
```
to increase the counter.

Reply:
Sure, thx for the suggestion. Will fix it.

Reply:
This is history problem, `ExecutionStatus ` need to split into two class `WorkflowExecutionStatus` and `TaskExecutionStatus`, some states are only belong to workflow, and some states are only belong to task, and some status are deprecated.

Reply:
> This is history problem, `ExecutionStatus ` need to split into two class `WorkflowExecutionStatus` and `TaskExecutionStatus`, some states are only belong to workflow, and some states are only belong to task, and some status are deprecated.

@ruanwenjun Considering the legacy here, what should we do in this PR? Maybe I could open a new issue to split this `ExecutionStatus `?

Reply:
I opened a new issue to split `ExecutionStatus` and remove deprecated states in it: #11145

Reply:
Another thing is if we use `ExecutionStatus.SUCCESS.getCode()` for tag, it might be a little bit unclear on the `Grafana` side as the tag value is integer in that case?